commit 3afef1d6cc0372086c44b5b1959811ac5b3d3ee7
Author: Christian Zietz <czietz@gmx.net>
Date:   Mon Dec 9 18:42:30 2019 +0000

    Get exception number from stack frame
    
    ... on CPUs (68010 and above) that support this.

diff --git a/bios/vectors.S b/bios/vectors.S
index 9b945334..b27344c7 100644
--- a/bios/vectors.S
+++ b/bios/vectors.S
@@ -238,6 +238,22 @@ any_vec:
         lsr.l   #1, d0
         add.l   #1, d0
         move.l  d0, _proc_enum.w
+
+#if CONF_WITH_ADVANCED_CPU
+        // D0 is 1 if the exception number could not be computed
+        subq    #1, d0
+        bne     dump_stack
+        // exception vector offset is present on stack for 68010 and above
+        tst.w   _longframe
+        beq     dump_stack
+        move.w  6(sp), d0
+        ext.l   d0
+        andi.l  #0xfff, d0
+        lsr.l   #2, d0
+        move.l  d0, _proc_enum.w
+#endif
+
+dump_stack:
         // dump 16 words from the stack
         movem.l (sp),d0-d7
 #ifdef __mcoldfire__
